addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1348,27 +1348,26 @@ const Wysiwyg = Widget.extend({
      * @param {Node} [params.htmlClass] Optionnal
      */
     openMediaDialog(params = {}) {
-        const sel = this.odooEditor.document.getSelection();
-
-        if (!sel.rangeCount) {
-            return;
-        }
-        const range = sel.getRangeAt(0);
         // We lose the current selection inside the content editable when we
         // click the media dialog button so we need to be able to restore the
         // selection when the modal is closed.
         const restoreSelection = preserveCursor(this.odooEditor.document);
```

Comment:
It's fine `restoreSelection` calls `setSelection` which has a guard in case the selection is null :)

https://github.com/odoo/odoo/blob/16.0/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L580-L589

Reply:
No. Before it does call `setSelection` it does:
```js
const sel = document.getSelection();
const cursorPos = [sel.anchorNode, sel.anchorOffset, sel.focusNode, sel.focusOffset];
```
If `sel` is `null` then the next line crashes.

Reply:
AFAIK we are in a browsing context so it's fine https://w3c.github.io/selection-api/#dom-document-getselection

Reply:
Well there seem to be at least one edge case in Firefox as described in https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection.

I don't know for sure if we do ever reach this particular case, or if there are other undocumented cases like this, but to be honest I don't understand the gain of removing the guarding lines that are already there. Did I miss a particular reason why those lines were an issue to begin with ?

Reply:
Shouldn't the guard be in preserveCursor() then? This way the calling code doesn't have to be aware of such edge case

Reply:
I thought about it, and it's debatable for sure. I mean, why do you want to preserve your cursor if you don't have a cursor to being with ? It's like you have a football player and you do like `player.kickTheBall()` except the player doesn't have the ball to being with... It could be argued that `kickTheBall` should be resilient in case it's being called when the player doesn't have the ball, but it could also be argued that the code sould not expect a player to be able to kick the ball, much less actually request them to kick it, if they actually do _not_ have the ball to begin with.

@rdeodoo do you have an opinion on the matter ?